<launch>
  <!-- Spawn the display server. -->
  <node name="ds_replay" pkg="freemovr_engine" type="display_server" args="--mode=virtual_world">
    <rosparam command="load" file="$(find strawlab_freeflight_experiments)/data/test/display_server_config.yaml" />
  </node>

  <node name="stimulus_mode" pkg="rostopic" type="rostopic" args="pub --latch stimulus_mode std_msgs/String StimulusCylinderAndModel" />
  <node name="model_filename" pkg="rostopic" type="rostopic" args="pub --latch model_filename std_msgs/String justpost1.osg" />

  <!-- Set the camera position. -->
  <node name="camera_position" pkg="rostopic" type="rostopic" args="pub --once -f $(find strawlab_freeflight_experiments)/data/test/camera_position.yaml /ds_replay/trackball_manipulator_state  freemovr_engine/TrackballManipulatorState">
  </node>

  <group>
    <remap from="pose" to="model_pose"/>
    <node name="set_model" pkg="freemovr_engine" type="send_pose" args="--xyz 0.2,0.3,0"/>
  </group>

</launch>
